test/ecto/multi_test.exs

```diff
@@ -356,6 +356,13 @@ defmodule Ecto.MultiTest do
     end
   end
 
+  test "add changeset with duplicate action" do
+    changeset = %{Changeset.change(%Comment{}) | action: :insert}
+    multi = Multi.new() |> Multi.insert(:changeset, changeset)
+
+    assert multi.operations == [{:changeset, {:changeset, %{changeset | action: :insert}, []}}]
```

Comment:
```suggestion
    assert multi.operations == [{:changeset, {:changeset, changeset, []}}]
```

?

Btw, why are we adding this test? It is missing in the test coverage?

Reply:
Yeah, from there, https://github.com/elixir-ecto/ecto/blob/master/lib/ecto/multi.ex#L408

```
defp put_action(%{action: action} = changeset, action) do
  changeset
end
```